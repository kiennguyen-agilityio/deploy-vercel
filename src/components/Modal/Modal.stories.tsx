import { StoryObj, Meta } from '@storybook/react';

import Modal from './Modal';

const meta: Meta<typeof Modal> = {
  component: Modal,
};

type Story = StoryObj<typeof Modal>;

export const Default: Story = {
  args: {
    isOpen: true,
    onClose: () => {},
    children: 'Modal Content',
  },
};

export default meta;
