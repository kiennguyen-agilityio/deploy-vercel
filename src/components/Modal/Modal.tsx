import { ReactNode } from 'react';

// style
import './Modal.css';

export interface Props {
  isOpen: boolean;
  onClose: () => void;
  children: ReactNode;
}

const Modal: React.FC<Props> = ({ isOpen, onClose, children }) => {
  if (!isOpen) {
    return null;
  }

  return (
    <div className="modal-overlay">
      <div className="modal-content">
        <div className="modal-header">
          <button className="modal-close-button" onClick={onClose} type="button">
            &times;
          </button>
        </div>
        <div className="modal-body">{children}</div>
      </div>
    </div>
  );
};

export default Modal;
