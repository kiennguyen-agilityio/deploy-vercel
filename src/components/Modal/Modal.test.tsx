import { fireEvent, render } from '@testing-library/react';
import Modal from './Modal';

const onCloseMock = jest.fn();
const setup = (props = {}) => {
  return render(
    <Modal isOpen={true} onClose={onCloseMock} {...props}>
      <div>Modal content</div>
    </Modal>
  );
};
describe('Modal component', () => {
  it('renders Modal component correctly', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('closes the modal when close button is clicked', () => {
    const { container } = setup();
    const closeButton = container.querySelector('.modal-close-button') as HTMLButtonElement;

    fireEvent.click(closeButton);

    expect(onCloseMock).toHaveBeenCalledTimes(1);
  });
});
