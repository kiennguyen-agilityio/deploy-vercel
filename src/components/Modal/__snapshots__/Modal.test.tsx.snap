// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders Modal component correctly 1`] = `
<div>
  <div
    class="modal-overlay"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <button
          class="modal-close-button"
          type="button"
        >
          Ã—
        </button>
      </div>
      <div
        class="modal-body"
      >
        <div>
          Modal content
        </div>
      </div>
    </div>
  </div>
</div>
`;
